resharper/resharper-unity/src/Rider/UnityRefreshTracker.cs

```diff
@@ -74,21 +77,24 @@ public Task Refresh(RefreshType force)
 
             task.ContinueWith(_ =>
             {
+                var list = new List<string> {solFolder.FullPath};
```

Comment:
Why ExecuteOrQueueEx? You can use TaskHost.GuardedMainThreadScheduler to continue on the main thread.